### ISSUE TITLE
Named pattern in Gupfile

### ISSUE THREAD

**Issue by np** [timbertson/gup#2]
Concretly I would like to name parts of the patterns in the Gupfile these variables should then be accessible through build script, using environment variables for instance.

Here is a mockup example:

```
build-html:
  %(base:*).html
build-archive:
  %(dir:*)/%(base:*).%(ext:tar.gz)
  %(dir:*)/%(base:*).%(ext:tar.xz)
```

If one builds foo/bar.html, in build-html one has access to `GUP_base=foo/bar`.
If one builds foo/bar.tar.gz, in build-archive one has access to `GUP_dir=foo`,
`GUP_base=bar`, and `GUP_ext=tar.gz`.

What do you think?

**Comment by lunakid** [comment]
+1 for simplified globbing syntax for the most common cases. But I'm also worried about the extra syntax clutter... Please come up with an ingeniously intuitive, light yet sufficiently general approach which can handle all the cases! ;)

**Comment by timbertson** [comment]
I'm thinking there should be a per-line pattern mode flag. It doesn't matter if it's a bit wordy, since you shouldn't need to write them that often. Something like:

builder.sh:
	/lit foo*bar
	/glob foo/*/
	/re ^foo/bar/baz
	/group %(base:*).html


I also like `=lit`, `=re` etc. `/` is good because it's not going to be present in a filename, but a leading ":" or "=" should be almost as rare (and they both work better as a mnemonic).

Thoughts?

**Comment by lunakid** [comment]
I'll need to actually start using gup first to have a feeling for that...
Till that, please verify that no such light syntax exist that can handle both (cases without an explicit prefix. I'd be a bit reluctant to add those new flag "keywords" to the language.
